### PR TITLE
Server: Add TTYServer, a rudimentary text console manager

### PR DESCRIPTION
**Pull request by deoxxa**
This should probably call out to a login program at some point. Right now
it just puts a root terminal on tty{1,2,3}.

Remember not to leave your Serenity workstation unattended!

### CODE REVIEW

File: Servers/TTYServer/main.cpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#include <AK/Assertions.h>
+#include <LibCore/CFile.h>
```
Comment (awesomekling) [inline Servers/TTYServer/main.cpp]: This doesn't get used, so we don't need to link with `-lcore` either :)

Comment (deoxxa) [reply Servers/TTYServer/main.cpp]: Guess who copy/pasted this from SystemServer? :D

**Comment by bugaevc** [comment]
Why does the kernel have do the spawning, instead of `SystemServer`?

Might be because there's no way to set `m_tty` from userspace ATM, but:
* `m_tty` doesn't affect much ATM, does it?
* let's add such a way! IIUC, just opening a TTY without `O_NOCTTY` should set it as that processes controlling terminal

**Comment by awesomekling** [comment]
> Why does the kernel have do the spawning, instead of `SystemServer`?

It doesn't have to :) It would be better to do it from `SystemServer`!

> Might be because there's no way to set `m_tty` from userspace ATM, but:

Yeah, basically, but we can fix this, like you say.

> * `m_tty` doesn't affect much ATM, does it?

Not terribly much! The kernel's TTY layer is far from complete :)

> * let's add such a way! IIUC, just opening a TTY without `O_NOCTTY` should set it as that processes controlling terminal

Yes, I think that's accurate.

**Pull request merged**